### PR TITLE
Contributing Stacking ensembles and making changes to Voting and Bagging ensembles

### PR DESCRIPTION
**Pull request by mfeffer**
Code here isn't necessarily final but the point of this pull request is to add and revise a few ensemble methods to make them compatible with Lale pipelines.

- Stacking: contributing `StackingClassifier` and `StackingRegressor` from scratch; overriding `_concatenate_predictions` 
to allow for pandas dataframe compatibility (sklearn by default converts to numpy arrays)
- Voting: contributing `VotingRegressor` from scratch and modifying `VotingClassifier` so that the latter has a `predict_proba` function (not exposed by default in sklearn but required by Lale post-mitigators)
- Bagging: contributing `BaggingRegressor` from scratch and modifying `BaggingClassifier` so that underlying estimators can receive pandas dataframes if provided (sklearn by default converts to numpy arrays)
- Tests added in `test_core_classifiers` and `test_core_regressors`

### CODE REVIEW

File: lale/lib/sklearn/bagging_classifier.py [hunk 1]
```diff
@@ -246,6 +317,41 @@
...
+    "properties": {
+        "X": {
+            "type": "array",
+            "items": {
+                "type": "array",
+                "items": {"type": "number"},
+            },
+            "description": "Test samples.",
+        },
+        "y": {
```
Comment (hirzel) [inline lale/lib/sklearn/bagging_classifier.py:333]: The schema for `y` is consistent with what we specified in fit and predict for BaggingClassifier. But it might be too restrictive, because in practice, class labels are often not numbers but strings or booleans. I recommend changing it to an `anyOf`, similar to what we have in many other classifiers such as [DecisionTreeClassifier](https://github.com/IBM/lale/blob/master/lale/lib/sklearn/decision_tree_classifier.py#L214-L221).

File: lale/lib/sklearn/stacking_classifier.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+                        ],
+                    },
+                    "description": "Base estimators which will be stacked together. Each element of the list is defined as a tuple of string (i.e. name) and an estimator instance. An estimator can be set to ‘drop’ using set_params.",
+                },
+                "final_estimator": {
+                    "anyOf": [{"laleType": "operator"}, {"enum": [None]}],
+                    "default": None,
+                    "description": "A classifier which will be used to combine the base estimators. The default classifier is a 'LogisticRegression'",
+                },
+                "cv": {
```
Comment (hirzel) [inline lale/lib/sklearn/stacking_classifier.py:81]: To resolve this TODO, I think we can specify the "cv" of StackingClassifier similarly to the "cv" of [Hyperopt](https://github.com/IBM/lale/blob/master/lale/lib/lale/hyperopt.py#L476).

File: lale/lib/sklearn/stacking_classifier.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+            "type": "array",
+            "items": {
+                "type": "array",
+                "items": {"type": "number"},
+            },
+            "description": "Training vectors, where n_samples is the number of samples and n_features is the number of features.",
+        },
+        "y": {
+            "type": "array",
+            "items": {"type": "number"},
```
Comment (hirzel) [inline lale/lib/sklearn/stacking_classifier.py:129]: The schema for `y` might be too restrictive, because in practice, class labels are often not numbers but strings or booleans. Instead, I'd make it an `anyOf`, like in [DecisionTreeClassifier](https://github.com/IBM/lale/blob/master/lale/lib/sklearn/decision_tree_classifier.py#L214-L221).

**Review by hirzel — APPROVED** [review]
Thanks a lot for this PR! It looks great overall. I left a few inline comments.

File: lale/lib/sklearn/ada_boost_classifier.py [hunk 4]
```diff
@@ -12,11 +12,14 @@
...
+import pandas as pd
 from sklearn.ensemble import AdaBoostClassifier as SKLModel
 
 import lale.docstrings
 import lale.operators
 
+from .function_transformer import FunctionTransformer
+
 
 class FitSpecProxy:
```
[the remark this answers is not in the document]
Comment (mfeffer) [reply lale/lib/sklearn/ada_boost_classifier.py:24]: Sure! Also moving it to a new file in case other operators end up needing it in the future.

File: lale/lib/sklearn/bagging_classifier.py [hunk 5]
```diff
@@ -12,11 +12,82 @@
...
 import lale.operators
 
+from .function_transformer import FunctionTransformer
+
+
+class _BaggingClassifierImpl:
+    def __init__(
+        self,
+        base_estimator=None,
+        n_estimators=10,
```
[the remark this answers is not in the document]
Comment (mfeffer) [reply lale/lib/sklearn/bagging_classifier.py:28]: Sure no problem.

File: lale/lib/sklearn/bagging_classifier.py (hunk 1 above)
Comment (mfeffer) [reply lale/lib/sklearn/bagging_classifier.py:333]: Makes sense, will do.
